syntax = "proto3";

service ObjectDetectionService {
    rpc DetectObjects (DetectionRequest) returns (DetectionResponse) {}
}

message DetectionRequest {
    string image_path = 1;
    string model_type = 2;  // Add this line for model type
}

message DetectionResponse {
    repeated DetectedObject objects = 1;
    float accuracy = 2;  // Add accuracy field
    string energy_efficiency = 3;  // Add energy efficiency field
    float cpu_usage = 4;  // Add CPU usage field
    float memory_usage = 5;  // Add memory usage field
    float throughput = 6;  // Add throughput field
    float energy_required = 7;  // Add energy required field
    float power_watts = 8;  // Add power watts field
    map<string, string> server_data = 9;  // Add server_data as a map
    float latency = 10;
}

message DetectedObject {
    string label = 1;
    float confidence = 2;
    int32 x = 3;
    int32 y = 4;
    int32 width = 5;
    int32 height = 6;
}
